src/parameters/add_parameters.jl

```diff
@@ -114,7 +125,10 @@ function add_parameters!(
     if !(ts_type <: Union{PSY.AbstractDeterministic, PSY.StaticTimeSeries})
         error("add_parameters! for TimeSeriesParameter is not compatible with $ts_type")
     end
-    ts_name = get_time_series_names(model)[T]
+    ts_names = get_time_series_names(model)
+    ts_name = ts_names[T]
+    time_series_multiplier_id =
+        findfirst(x -> x == T, [first(p) for p in ts_names if last(p) == ts_name])
```

Comment:
@daniel-thom and @jd-lara is this OK?
It will return 1 for everything where there is only one occurrence of a time series name in the dict. If there are more than one entries referencing a time series name, it will return different values for each parameter.

Reply:
Can you describe what you are looking to achieve? I don't immediately understand the logic.

Reply:
In principle, nice solution. I didn't know that ts_names was a dict, and so I was tripped up at first. Is it more clear when written as
```
[k for (k, v) in ts_names if v == ts_name]
```
Secondly, since the logic is non-trivial and you repeat it, I think it would be good to put it in a function.

Per Jose's comment, perhaps the name of the function could serve as a comment.